packages/instant-apps-components/src/components/instant-apps-export/instant-apps-export.tsx

```diff
@@ -562,22 +575,29 @@ export class InstantAppsExport {
 
   async viewScreenshot(): Promise<void> {
     if (this.view != null && this.includeMap) {
-      const screenshot = await this.view.takeScreenshot({
-        width: this.view.width,
-        height: this.view.height,
-      });
-      const maxIMGWidth = 800;
-      const maxIMGHeight = 494;
+      if (this.screenshot == null) {
+        this.screenshot = await this.view.takeScreenshot({
+          width: this.view.width * 2,
+          height: this.view.height * 2,
+        });
+      }
+      this.handleScaleBarSize();
       if (this.viewEl != null && this.viewWrapperEl != null) {
-        if (this.view.width < maxIMGWidth) {
-          this.viewWrapperEl.style.width = `${this.view.width}px`;
-          this.viewWrapperEl.style.maxWidth = '';
+        const { height, width } = this.screenshot.data;
+        if (height > width) {
```

Comment:
can the logic in the if/else blocks be broken up into separate methods, with method names that describe what each block is doing?